rclpy/rclpy/executors.py

```diff
@@ -802,10 +802,13 @@ def _spin_once_impl(
         else:
             self._executor.submit(handler)
             self._futures.append(handler)
-            for future in self._futures:  # check for any exceptions
+            self._futures.append(handler)
+            # make a copy of the list that we iterate over while modifying it
+            # (https://stackoverflow.com/q/1207406/3753684)
+            for future in self._futures[:]:
                 if future.done():
                     self._futures.remove(future)
```

Comment:
Just a comment to share. this will be slower, since it has to scan the list every iteration to remove.

Reply:
Jup, true. That's why I also commented on it, to explain why you'd do something like this.

Reply:
One could run some benchmarks to see if it affects performance. I don't know whether this is done regularly. 🤷 I don't plan on doing this.

Reply:
that was just a comment, i do not think we need to benchmark test for this.